### DISCUSSION TITLE
Feedback / Feature suggestions

### DISCUSSION THREAD

**Discussion by ovizii** [docusealco/docuseal#94] [Ideas]
I'd like to thank you for this awesome tool. I am using it privately and it works jsut fine for me. 

However I have a few suggestions to make:

- implement some kind of workflow to make sure submitters get access in a specific order
- rethink the usage of the word "templates" - I find it confusing as when I think of templates, I expect something I can reuse later and can't use directly. As it is now, I create a template and then add recipients and start using this template. (Let me know if this was not clear enough, I could try to be more clear)
- I would love to have a feature for saving my signature, say the first time I sign, I'd use a touchpad and then save the signature, so I can reuse it for when I have no touch capable input device connected.
- I would really appreciate the addition of a once-off option to sign documents for "self-usage" without any recipients. I often find myself needing to sign some documents i.e. cancelling a credit card or similar where I then need to somehow add my signature scribble to a document. Docuseal is perfect for this, but these are once-off so having a separate template for it is overkill.

**Comment by AlexBTurchyn** [comment]
@ovizii We appreciate your interest in DocuSeal. Regarding your suggestions:
> implement some kind of workflow to make sure submitters get access in a specific order

It's in the roadmap already - will be released eventually

> rethink the usage of the word "templates" - I find it confusing as when I think of templates, I expect something I can reuse later and can't use directly. As it is now, I create a template and then add recipients and start using this template. (Let me know if this was not clear enough, I could try to be more clear)

You are right that it can confuse someone, but using more specific words can have an even more negative effect. It would be great if you could suggest some alternative

> I would love to have a feature for saving my signature, say the first time I sign, I'd use a touchpad and then save the signature, so I can reuse it for when I have no touch capable input device connected.

We can store signatures only for admin users (in case they cross-sign docs) - signers that are using DocuSeal via the signing form don't have their own account/user record to store some additional info about them between sessions. Did you mean to have this implemented for admin users?

> I would really appreciate the addition of a once-off option to sign documents for "self-usage" without any recipients. I often find myself needing to sign some documents i.e. cancelling a credit card or similar where I then need to somehow add my signature scribble to a document. Docuseal is perfect for this, but these are once-off so having a separate template for it is overkill.

You're right, you can self-sign docs via a one-off template - that's the only option for now. Usually people use desktop apps if they need to sign docs by themselves so bringing this feature into DocuSeal won't bring a lot of value for the majority of users. But still thats an interesting feature and hopefully it will be added some day.

**Reply by ovizii** [reply]
Thanks for taking a look and replying, much appreciated. 

Regarding:
- workflow: great to hear its on the roadmap
- not sure about a better term for templates, maybe keep the template part but disallow adding recipients or submitting it myself? This would force them to be used as templates. Then, once you clone it, it will not be cloned into a template but into a maybe project? Anyway, these are just suggestions, possibly they help, possibly they don't.
- can you explain how/where me as an admin can store my signature? I can't seem to figure that feature out.
- regarding once-off: I understand your point, I also used a desktop app but when switching from OSX to Windows I had to find a different solution and docuseal sounded great, but I understand it is a niche-case :-)

**Reply by AlexBTurchyn** [reply]
>can you explain how/where me as an admin can store my signature? I can't seem to figure that feature out.

it's not ready yet :) but definitely that's an important feature to add to the project

**Reply by ovizii** [reply]
Let me add a quick follow-up:

I just realized that "delete" actually archives and I can't figure out how to really delete :-( 
Also, if you have no active templates the link top the archive disappears. 

I really love this tool, but I often need it just for myself to fill and sign a few plain docs, email them to the recipient and delete all traces of them as I'll never need them again.

**Reply by omohokcoj** [reply]
@ovizii you can now permanently delete documents/templates in the latest docuseal version

**Reply by ste-edu** [reply]
@AlexandrToorchyn could you please share any roadmap or ETA for this feature:

    **_**implement some kind of workflow to make sure submitters get access in a specific order**_**
    
This is also discussed in other threads (e.g. reddit) and I think this particular addition (a signing order) would make Docuseal an ideal candidate for lots of use cases. 

Thanks in advance for the info.

**Reply by omohokcoj** [reply]
@ste-edu tis feature has been added recently - you can decide if all parties receive signature request emails right away, or in the original order when the next signer receives their signature request email only if the document was signed by the previous signer:
![image](https://github.com/docusealco/docuseal/assets/5418788/be7c0a74-dca6-4c68-a552-78fb56bf91f4)
